audinterface/core/feature.py

```diff
@@ -415,6 +417,92 @@ def to_numpy(
         """
         return frame.values.T.reshape(self.num_channels, self.num_features, -1)
 
+    def _assert_shape_3d(
+            self,
+            features: typing.Union[np.ndarray, pd.Series]
+    ):
+        r"""Reshape to [n_channels, n_features, n_frames]."""
+
+        if self.process.process_func_is_mono:
+            features = np.array(features)
+            # when mono processing is turned on
+            # the channel dimension has to be 1
+            # so we would usually omit it,
+            # but since older versions required
+            # a channel dimension we have to
+            # consider two special cases
+            if (features.ndim == 4) and \
+                    (features.shape[1] == 1):
+                # (1, features, frames)
+                # -> (channels, 1, features, frames)
+                # -> (channels, features, frames)
```

Comment:
This comment is slightly misleading.
What you are trying to do here is to fix
```
(channels, channels, features, frames)
```
to
```
(channels, features, frames)
```
correct?

Reply:
I also don't complete get, why we can have `(channels, channels, features, frames)` in the first place?

Reply:
It's for the special case that `process_func_is_mono=True`. In that case we get a list of `numpy.arrays` we need to combine to an array of shape `(channels, features, frames)`. Now, when a user sets `process_func_is_mono=True` it makes sense to return `(features, frames)` since the process function only gets mono audio. But of course we don't want to forbid that it returns `(1, features, frames)` (actually, before this MR it even had to return this shape). But that means that when we combine the arrays we get `(channels, 1, features, frames)`, so in that case we have to remove the second dimension.

Reply:
Thanks for the detailed explanation.

Then maybe change the comment to:

```python
# (1, features, frames)  mono feature extractor
# -> (channels, 1, features, frames)  combined extracted mono features
# -> (channels, features, frames)  desired shape
```

or

```python
# (channels, 1, features, frames)
# -> (channels, features, frames)
```

Reply:
Ok, changed it accordingly.